resources/widgetlist.ts

```diff
@@ -238,34 +274,33 @@ class WidgetList extends HTMLElement {
     this._nameToId[name] = id;
     this._elements[id] = sortKeyFn;
     this._sorted.push(id);
-    const that = this;
     this._sorted.sort((a, b) => {
-      return that._elements[a]() - that._elements[b]();
+      return this._elements[a]() - this._elements[b]();
     });
 
     element.style.position = 'relative';
-    element.style.left = element.style.top = 0;
+    element.style.left = element.style.top = '0';
 
     const container = document.createElement('div');
     container.appendChild(element);
     container.id = 'child' + id;
 
-    this.rootElement.appendChild(container);
+    this.rootElement!.appendChild(container);
```

Comment:
Why are here and everywhere using `!.` oprand instead of a explicit null-check?

Reply:
I just translation what old js do into ts. And in this case, root element won't be null.